models/season.js

```diff
@@ -112,10 +112,108 @@ const SeasonSchema = new mongoose.Schema({
   teams: [SeasonTeamSchema],
   drivers: [SeasonDriverSchema],
 }, { collection: 'seasons' })
-  .plugin(normalizeJSON)
   .plugin(timestamps)
   .index({ name: 1, championship: 1 }, { unique: true });
 
+SeasonSchema.set('toJSON', {
+  transform: async (doc, json) => {
+    const now = Date.now();
+
+    /* eslint-disable no-param-reassign, no-underscore-dangle */
+    json.id = json._id;
+    delete json._id;
+    delete json.__v;
+    /* eslint-enable no-underscore-dangle */
+
+    await Promise.all([
+      doc.populate('teams.team').execPopulate(),
+      doc.populate('drivers.driver').execPopulate(),
+      doc.populate('grandsPrix.circuit').execPopulate(),
+      doc.populate('grandsPrix.grandPrix').execPopulate(),
+    ]);
+
+    json.drivers = [];
+    json.teams = [];
+    json.grandsPrix = [];
+
+    const grandsPrix = doc.grandsPrix.sort((gp1, gp2) => gp1.raceUTC - gp2.raceUTC);
+
+    const nextGrandPrix = grandsPrix.find(gp => gp.raceUTC > now);
+    const previousGrandPrix = [...grandsPrix].reverse().find(gp => gp.raceUTC < now);
+
+    json.marketOpen = nextGrandPrix && (!previousGrandPrix || previousGrandPrix.results.length > 0);
+    json.currentTransactionFeePercentage = 0;
+
+    if (json.marketOpen) {
+      if (now > nextGrandPrix.qualifyingUTC) {
```

Comment:
(question) would it be better with switch/case? (waiting for your look over the monitors...xD)